components/feed/LikeButton.tsx

```diff
@@ -1,60 +1,102 @@
 "use client";
 
-import React, { useEffect, useState } from "react";
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
 
-const LikeButton = ({ filled = false }: { filled?: boolean }) => {
-  const [state, setState] = useState(filled);
+import { formatFollowers } from "@/lib/formatters";
+import { likeItem, unlikeItem } from "@/lib/likeButton";
 
-  useEffect(() => {
-    // eslint-disable-next-line no-console
-    console.log("like change");
-  }, [state]);
+const LikeButton = ({
+  userId,
+  postId,
+  likes,
+  likedIds,
+}: {
+  userId: string;
+  postId: string;
+  likes: number;
+  likedIds: string[] | undefined;
+}) => {
+  const liked = likedIds?.includes(userId) ?? false;
 
-  const onClick = () => {
-    if (!state) {
-      setState(true);
+  // Created this "local" state to fix how long it would take between
+  //  clicking the like button and like button being updated
+  // Then still updates to current with database
+  const [localLikesNum, setLocalLikesNum] = useState(likes);
+  const [localLikedState, setLocalLikedState] = useState(liked);
+
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const [isFetching, setIsFetching] = useState(false);
+
+  // Create inline loading UI
+  const isMutating = isFetching || isPending;
+
+  const handleClick = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    setIsFetching(true);
+    if (localLikedState) {
+      setLocalLikesNum(localLikesNum - 1);
+      setLocalLikedState(!localLikedState);
+      await unlikeItem(userId, postId);
     } else {
-      setState(false);
+      setLocalLikesNum(localLikesNum + 1);
+      setLocalLikedState(!localLikedState);
+      await likeItem(userId, postId);
     }
+    setIsFetching(false);
+    startTransition(() => {
+      // Refresh the current route and fetch new data from the server without
+      // losing client-side browser or React state.
+      router.refresh();
+    });
   };
 
-  if (state) {
+  if (!localLikedState) {
     return (
+      <div className="flex flex-row">
+        <svg
+          onClick={(e) => void handleClick(e)}
+          fill="black"
```

Comment:
Are the two svgs identical except for the fill? Can you delete the if statement for localLikedState and just have one unified return? something like 
```suggestion
          fill={localLikedState ? "white" : "black"}
```